### DISCUSSION TITLE
Toggle light and dark mode programmatically

### DISCUSSION THREAD

**Discussion by Lyokolux** [inkline/inkline#311] [Need help] [answered]
Is there a way to update the dark mode on a button click.

Here the default dark mode:
<img width="742" alt="Bildschirmfoto 2022-09-10 um 18 41 38" src="https://user-images.githubusercontent.com/29046067/189493195-22c916f6-9ebd-4c76-a0f9-8b64cb914594.png">

When the button is triggered, the body class `-dark` is removed. Some UI elements stays dark though:
<img width="741" alt="Bildschirmfoto 2022-09-10 um 18 41 42" src="https://user-images.githubusercontent.com/29046067/189493205-b16dae8c-84ff-4a7c-a112-f1260642d6bd.png">

The class `-dark` is applied on multiple components.  
The workaround I found is to parse every node containing the class (`document.querySelectorAll('.-dark')`), then remove it for each node.
Something like:
```ts
watchEffect(() => {
  if (isDark.value === false) {
    document.querySelectorAll('.-dark').forEach((node) => node.classList.remove('-dark'))
  } else {
    document.querySelectorAll('[class*=input-wrapper]').forEach((node) => node.classList.add('-dark'))
    document.querySelectorAll('[class*=button]').forEach((node) => node.classList.add('-dark'))
  }
})
````

Is there a better way?

**Comment by alexgrozav** [comment, marked as answer]
Hey @Lyokolux! Can you provide a small reproduction repo so I can look further into this?

**Reply by alexgrozav** [reply]
Also, have you followed this guide for the button's functionality?

https://www.inkline.io/docs/introduction/guides/dark-mode

**Reply by Lyokolux** [reply]
I didn't follow it as I missed it in the documentation. My bad. This discussion can be closed. Thanks for it!

**Reply by alexgrozav** [reply]
No worries! Glad I could help :)
